File: API/API/Models/TodoTask.vb
Imports System.Threading.Tasks

Public Class TodoTask

    Public Property Id As Integer

    Public Property Title As String
    Public Property Description As String

    Public Property CreatedAt As DateTime
    Public Property Deadline As DateTime?

    Public Property Status As TaskStatus


    Public Property CreatorId As Guid

    Public Overridable Property Creator As User

    Public Overridable Property Assignments As ICollection(Of TaskAssignment)

    Public Property ProjectId As Guid?
    Public Overridable Property Project As Project

End Class
Public Enum TaskStatus
    ToDo = 0
    InProgress = 1
    Done = 2
    Closed = 3
End Enum

File: API_v2/Models/TodoTask.cs
using System;
using System.Collections.Generic;
using API_v2.Models.Enums;

namespace API_v2.Models
{
    public class TodoTask
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? Deadline { get; set; }
        public DateTime? StartDate { get; set; }
        public double? EstimatedHours { get; set; }
        public double? ActualHours { get; set; }
        public int ColumnId { get; set; }
        public Guid CreatorId { get; set; }
        public Guid? ProjectId { get; set; }
        public TaskPriority Priority { get; set; } = TaskPriority.Medium;

        // Navigation properties
        public virtual User Creator { get; set; } = null!;
        public virtual Project? Project { get; set; }
        public virtual ProjectColumn Column { get; set; } = null!;
        public virtual ICollection<TaskAssignment> Assignments { get; set; } = new List<TaskAssignment>();
        public virtual ICollection<TaskComment> Comments { get; set; } = new List<TaskComment>();
    }
}
